src/libraries/System.Private.CoreLib/src/System/Globalization/TextInfo.cs

```diff
@@ -528,11 +538,13 @@ internal static void ToUpperAsciiInvariant(ReadOnlySpan<char> source, Span<char>
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static char ToLowerAsciiInvariant(char c)
         {
-            if ((uint)(c - 'A') <= (uint)('Z' - 'A'))
+            if (UnicodeUtility.IsInRangeInclusive(c, 'A', 'Z'))
             {
-                c = (char)(c | 0x20);
+                // on x86, extending BYTE -> DWORD is more efficient than WORD -> DWORD
+                c = (char)(byte)(c | 0x20);
```

Comment:
I'm not understanding the comment.  You're saying:
```C#
c = (char)(byte)(c | 0x20);
```
results in better asm than:
```C#
c = (char)(c | 0x20);
```
?

Reply:
It's
```diff
- movzx ecx, ax
+ movzx ecx, al
``` 
so one of the most low level optimizations 😅 

If this does matter (tbh: I doubt it), then see https://github.com/dotnet/runtime/pull/35194#discussion_r411212802

Reply:
Working with 16-bit data types is weirdly expensive on x86 processors. Working with 8-bit and 32-bit data types is cheaper. Zero-extending 8-bit to 32-bit is occasionally free.

Reply:
For clarification, a movzx of 8-bit to 32-bit can sometimes be elided entirely by the processor. In the worst case it's a 1/4 nanosecond latency on modern CPUs.

A movzx of 16-bit to 32-bit can never be elided by the processor. In the worst case it's a multi-cycle latency, as you have the cost of the operation _and_ the fact that this instruction operates on a dedicated port separate from many other ALU instructions. So that's a potential pipeline stall that affects anybody who will end up reading this value after it's computed.